### PR TITLE
openstack: Add CCM job

### PR DESCRIPTION
**Pull request by mdbooth**

### CODE REVIEW

**Comment by mandre** [comment]
This is interesting... First time I see that:
```
level=fatal msg=failed to fetch Terraform Variables: failed to generate asset "Terraform Variables": failed to get openstack Terraform variables: flate: corrupt input before offset 95
```
/test pj-rehearse

**Comment by mandre** [comment]
We'll need more logs in `ccm-conf-apply-feature-gate-commands.sh` to understand why `setExternalFlagKCMO()` doesn't work on openstack platform.

**Comment by mdbooth** [comment]
> 
> 
> We'll need more logs in `ccm-conf-apply-feature-gate-commands.sh` to understand why `setExternalFlagKCMO()` doesn't work on openstack platform.

Also need to understand why it's taking the full 2 hours to timeout rather than the intended 30m.

**Comment by mandre** [comment]
> > We'll need more logs in `ccm-conf-apply-feature-gate-commands.sh` to understand why `setExternalFlagKCMO()` doesn't work on openstack platform.
> 
> Also need to understand why it's taking the full 2 hours to timeout rather than the intended 30m.

Yes, the timeout implementation is not working but that's a separate issue.

File: ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-master.yaml [hunk 1]
```diff
@@ -68,16 +68,16 @@ tests:
...
-    env:
-      TEST_SUITE: experimental/reliability/minimal
-    workflow: openshift-e2e-openstack-ipi
 - as: e2e-aws-ccm-install
   steps:
     cluster_profile: aws
     workflow: openshift-e2e-aws-ccm-install
+- as: e2e-openstack-ccm
+  steps:
+    cluster_profile: openstack-vexxhost
```
Comment (Fedosin) [inline ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-master.yaml:77]:
```suggestion
    cluster_profile: openstack
```

Let's try it for now

Comment (mdbooth) [reply ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-master.yaml]: I'm trying vh-mecha. MOC is sadness. I don't want to run anything there.

**Comment by mdbooth** [comment]
The fix in console-operator only just merged so it may be too soon, but...

/test pj-rehearse

**Comment by mdbooth** [comment]
This is now failing in etcd:

```
#### attempt 0
      member={name="b56h8ssw-433ed-8xt5r-master-1", peerURLs=[https://10.0.1.224:2380}, clientURLs=[https://10.0.1.224:2379,unixs://10.0.1.224:0]
      member={name="b56h8ssw-433ed-8xt5r-master-0", peerURLs=[https://10.0.2.171:2380}, clientURLs=[https://10.0.2.171:2379,unixs://10.0.2.171:0]
      member={name="b56h8ssw-433ed-8xt5r-master-2", peerURLs=[https://10.0.1.143:2380}, clientURLs=[https://10.0.1.143:2379,unixs://10.0.1.143:0]
      target={name="b56h8ssw-433ed-8xt5r-master-2", peerURLs=[https://10.0.1.143:2380}, clientURLs=[https://10.0.1.143:2379,unixs://10.0.1.143:0]
member "https://10.0.1.143:2380" dataDir has been destroyed and must be removed from the cluster[mbooth@DESKTOP-07R2QD0 tmp]$
```

This will be because we use etcd-on-ramfs on OpenStack, and applying the CCM featuregate reboots the node. I'll look into this on Monday.

**Comment by EricArrakis** [comment]
> This is now failing in etcd:
> 
> ```
> #### attempt 0
>       member={name="b56h8ssw-433ed-8xt5r-master-1", peerURLs=[https://10.0.1.224:2380}, clientURLs=[https://10.0.1.224:2379,unixs://10.0.1.224:0]
>       member={name="b56h8ssw-433ed-8xt5r-master-0", peerURLs=[https://10.0.2.171:2380}, clientURLs=[https://10.0.2.171:2379,unixs://10.0.2.171:0]
>       member={name="b56h8ssw-433ed-8xt5r-master-2", peerURLs=[https://10.0.1.143:2380}, clientURLs=[https://10.0.1.143:2379,unixs://10.0.1.143:0]
>       target={name="b56h8ssw-433ed-8xt5r-master-2", peerURLs=[https://10.0.1.143:2380}, clientURLs=[https://10.0.1.143:2379,unixs://10.0.1.143:0]
> member "https://10.0.1.143:2380" dataDir has been destroyed and must be removed from the cluster[mbooth@DESKTOP-07R2QD0 tmp]$
> ```
> 
> This will be because we use etcd-on-ramfs on OpenStack, and applying the CCM featuregate reboots the node. I'll look into this on Monday.

If we are running on mecha, do we still need ramfs?  Don't we have access to NVMe, and therefore etcd should be happier?

**Comment by mdbooth** [comment]
> 
> 
> > This is now failing in etcd:
> > ```
> > #### attempt 0
> >       member={name="b56h8ssw-433ed-8xt5r-master-1", peerURLs=[https://10.0.1.224:2380}, clientURLs=[https://10.0.1.224:2379,unixs://10.0.1.224:0]
> >       member={name="b56h8ssw-433ed-8xt5r-master-0", peerURLs=[https://10.0.2.171:2380}, clientURLs=[https://10.0.2.171:2379,unixs://10.0.2.171:0]
> >       member={name="b56h8ssw-433ed-8xt5r-master-2", peerURLs=[https://10.0.1.143:2380}, clientURLs=[https://10.0.1.143:2379,unixs://10.0.1.143:0]
> >       target={name="b56h8ssw-433ed-8xt5r-master-2", peerURLs=[https://10.0.1.143:2380}, clientURLs=[https://10.0.1.143:2379,unixs://10.0.1.143:0]
> > member "https://10.0.1.143:2380" dataDir has been destroyed and must be removed from the cluster[mbooth@DESKTOP-07R2QD0 tmp]$
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This will be because we use etcd-on-ramfs on OpenStack, and applying the CCM featuregate reboots the node. I'll look into this on Monday.
> 
> If we are running on mecha, do we still need ramfs? Don't we have access to NVMe, and therefore etcd should be happier?

There's no NVMe on mecha, but it does have nice fast SSDs which... we're accessing via local ceph :(

This has been fun: https://github.com/shiftstack/dev-install/pull/131. Hopefully get that change rolled out today.

**Review by Fedosin — APPROVED** [review]
/lgtm
/hold

**Review by Fedosin — COMMENTED** [review]
/lgtm

**Pull request merged**